src/ec2/model.ts

```diff
@@ -41,21 +42,33 @@ export class Ec2ConnectionManager {
         return new DefaultIamClient(this.regionCode)
     }
 
-    protected async getAttachedPolicies(instanceId: string): Promise<IAM.attachedPoliciesListType> {
+    protected async getAttachedPolicies(instanceId: string): Promise<IAM.AttachedPolicy[]> {
         const IamRole = await this.ec2Client.getAttachedIamRole(instanceId)
-        if (!IamRole) {
+        if (!IamRole?.Arn) {
             return []
         }
-        const iamResponse = await this.iamClient.listAttachedRolePolicies(IamRole!.Arn!)
-
-        return iamResponse.AttachedPolicies ?? []
+        try {
+            const attachedPolicies = await this.iamClient.listAttachedRolePolicies(IamRole!.Arn!)
```

Comment:
does the if-check above avoid this?

```suggestion
            const attachedPolicies = await this.iamClient.listAttachedRolePolicies(IamRole.Arn)
```